ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.BusinessLogic/Validators/Abstract/BaseTaskDTOValidator.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace ExadelTimeTrackingSystem.BusinessLogic.Validators.Abstract
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using ExadelTimeTrackingSystem.BusinessLogic.DTOs;
+    using ExadelTimeTrackingSystem.BusinessLogic.Services.Abstract;
+    using FluentValidation;
+
+    public abstract class BaseTaskDTOValidator<TDTO> : AbstractValidator<TDTO>
+        where TDTO : CreateTaskDTO
+    {
+        private readonly IUserService _userService;
+        private readonly IProjectService _projectService;
+
+        public BaseTaskDTOValidator(IUserService userService, IProjectService projectService)
+        {
+            _userService = userService;
+            _projectService = projectService;
+            ConfigureRules();
+        }
+
+        private void ConfigureRules()
+        {
+            RuleFor(t => t.HoursSpent)
+                .InclusiveBetween(0, 1000);
+
+            RuleFor(t => t.ApproverId)
+                .NotEmpty()
+                .MustAsync(_userService.ExistsAsync)
+                .WithMessage(Constants.Validation.APPROVER_ID_DOES_NOT_EXIST);
+
+            RuleFor(t => t.Date)
+                .NotEmpty();
+
+            RuleFor(t => t.Activity)
```

Comment:
Check that activity exists for the project for which task is created

Reply:
Remember for each project we defined list of allowed activities. This Activity of task should not be random, but should be allowed for project